projects/netgrif-components-core/src/lib/authorization/role/role-guard.service.ts

```diff
@@ -37,22 +37,40 @@ export class RoleGuardService implements CanActivate {
     }
 
     public canAccessView(view: View, url: string): boolean {
-        if (typeof view.access !== 'string' && view.access.hasOwnProperty('role')) {
-            const allowedRoles = this.parseRoleConstraints(view.access.role, url);
-
-            return allowedRoles.some(constraint => {
-                if (constraint.roleIdentifier) {
-                    return this._userService.hasRoleByIdentifier(constraint.roleIdentifier, constraint.processIdentifier);
-                } else {
-                    return this._userService.hasRoleByName(constraint.roleName, constraint.processIdentifier);
-                }
-            });
+        if (typeof view.access !== 'string' && (view.access.hasOwnProperty('role') || view.access.hasOwnProperty('bannedRole'))) {
+            let result = false;
+
+            if (view.access.hasOwnProperty('role')) {
+                const allowedRoles = this.parseRoleConstraints(view.access.role, url);
+
+                result = allowedRoles.some(constraint => {
+                    if (constraint.roleIdentifier) {
+                        return this._userService.hasRoleByIdentifier(constraint.roleIdentifier, constraint.processIdentifier);
+                    } else {
+                        return this._userService.hasRoleByName(constraint.roleName, constraint.processIdentifier);
+                    }
+                });
+            }
+
+            if (view.access.hasOwnProperty('bannedRole')) {
```

Comment:
what is the role policy for menu entries?

if the menu entry has both the `role` and the `bannedRole` property, the current implementation will display the menu entry if the user has none of the roles and none of the banned roles. Is this intended?

Reply:
Done